src/freertos_drivers/esp32/Esp32WiFiManager.hxx

```diff
@@ -0,0 +1,316 @@
+    /// @param hubHostname is the hostname for the remote hub.
+    /// @param ip is the remote hub ip address.
+    /// @param remotePort is the port on the hub to connect to.
+    /// @return true if the connection to the hub was successful or if already
+    /// connected to that hub, false otherwise.
+    bool connectToHub(String hubHostname, IPAddress ip, uint16_t remotePort);
+
+    /// Attempts to connect to a manually configured hub if this node is
+    /// configured to do so.
+    /// @return true if we successfully connected to the hub or if we need
+    /// to try mDNS.
+    bool connectToManualHubIfNeeded();
+
+    /// Attempts to connect to any mDNS identified hubs that we are not already
+    /// connected to.
+    /// @return true if we successfully connected to at least one hub, false
+    /// otherwise.
+    bool connectToMDNSHubs();
+
+    /// Priority to use for the wifi_manager_task. This is currently set to 1
+    /// which matches the arduino-esp32 loop() task priority.
+    static constexpr UBaseType_t WIFI_TASK_PRIORITY = 1;
+
+    /// Stack size for the wifi_manager_task.
+    static constexpr uint32_t WIFI_TASK_STACK_SIZE = 2048L;
+
+    /// Handle for the wifi_manager_task that manages the WiFi stack, including
+    /// periodic health checks of the connected hubs or clients.
+    TaskHandle_t wifiTaskHandle_;
+
+    /// Map of connections to any hubs that are to be maintained.
+    std::map<Esp32WiFiClientAdapter *, Executable *> hubConnections_;
+
+    /// Map of connections from other nodes to be monitored.
+    std::map<Esp32WiFiClientAdapter *, Executable *> clientConnections_;
+
+    /// Internal flag to tell the wifi_manager_task to shutdown.
+    bool wifiTaskShutdownReq_{false};
+
+    /// Internal flag used by the wifi_manager_task to indicate it is running.
+    bool wifiTaskRunning_{false};
+
+    /// Dynamically generated hostname for this node.
+    String hostname_;
+
+    /// User provided SSID to connect to.
+    const char *ssid_;
+
+    /// User provided password for the SSID to connect to.
+    const char *password_;
+
+    /// Persistent configuration that will be used for this node's WiFi usage.
+    const WiFiConfiguration cfg_;
+
+    /// This is internally used to enable the management of the WiFi stack, in
+    /// some environments this may be managed externally.
+    bool manageWiFi_;
+
+    /// OpenMRN stack for the Arduino system
+    OpenMRN *openmrn_;
+
+    /// Cached copy of this nodes wifi_sleep value. If this is true the WiFi
+    /// radio can go to into low power mode to conserve power. This is likely
+    /// only necessary for battery powered ESP32 devices.
+    bool wifiSleepMode_{false};
+
+    /// Cached copy of this nodes hub_config.enabled value, if this is
+    /// true this node will advertise itself as a hub via mDNS and accept
+    /// incoming TCP/IP connections.
+    bool hubEnabled_{false};
+
+    /// Cached copy of this nodes hub_config.port
```

Comment:
The only ones that would be needed likely will be the IPAddress/port for manual and service name values since the esp32 wants them split rather than as a complete name. let me look at dropping these.